pages/index.js

```diff
@@ -1,14 +1,49 @@
 import ActivityCard from "@/components/Activitylist/ActivityCard";
 import ActivityList from "@/components/Activitylist/ActivityList";
+import useSWR from "swr";
+import { useState } from "react";
 
+export default function HomePage() {
+  const { data: activities, isLoading, error } = useSWR("/api/activities");
+  const [search, setSearch] = useState("");
 
+  if (isLoading) return <p>Loading...</p>;
+  if (error) return <p>Error to load the Page.</p>;
+
+  const filterActivities = activities.filter((activity) => {
+    const match = search.toLowerCase();
+
+    const title = activity.title?.toLowerCase() || "";
+    const category = activity.category?.toLowerCase() || "";
+    const country = activity.country?.toLowerCase() || "";
+
+    return (
+      title.includes(match) ||
+      category.includes(match) ||
+      country.includes(match)
+    );
+  });
+
+  function handleSearch(event) {
+    const inputElement = event.target;
+    const newValue = inputElement.value;
+
+    setSearch(newValue);
+  }
 
-export default function HomePage() {
   return (
     <>
       <h1>Activity Planner</h1>
       <h2>for your next journey</h2>
-      <ActivityList />
+      <input
+        placeholder="Filter: Title, Category, Country"
+        value={search}
+        onChange={handleSearch}
+      />
+      <button type="button" onClick={() => setSearch("")}>
+        Remove
+      </button>
```

Comment:
please move this into a separate Component and wrap it with a form element, also move the event.target logic into that component like this: 

```suggestion
      <input
        placeholder="Filter: Title, Category, Country"
        value={search}
        onChange={(event) => handleSearch(event.target.value)}
      />
      <button type="button" onClick={() => handleSearch("")}>
        Remove
      </button>
```